workflow/controller/operator.go

```diff
@@ -1265,9 +1255,43 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 	return nil
 }
 
-func getExitCode(pod *apiv1.Pod) *int32 {
+func (woc *wfOperationCtx) updateNodeExitData(pod *apiv1.Pod, node *wfv1.NodeStatus) *wfv1.NodeStatus {
+	updated := false
+	containerName := node.GetContainerName()
+	if exitCode := getExitCode(pod, containerName); exitCode != nil {
```

Comment:
interesting --- this was  a bug

Reply:
the existing code is inconsistent - some of it update variables like `newPhase` and some directly updates `node.Phase` - this will cause bugs (e.g. when updating node.Phase, we don't set updated=true)

I think we need to refactor it - but not in this PR